fastjet/internal/delaunay/point.go

```diff
@@ -50,13 +51,21 @@ func (p *Point) inTriangle(t *Triangle) (inside, edge bool) {
 	barcen2 := ((t.C.Y-t.A.Y)*(p.X-t.C.X) + (t.A.X-t.C.X)*(p.Y-t.C.Y)) / det(t)
 	barcen3 := 1 - barcen1 - barcen2
 
-	// inside triangle
-	if barcen1 > 0 && barcen1 < 1 && barcen2 > 0 && barcen2 < 1 && barcen3 > 0 && barcen3 < 1 {
+	// use the math/big package to handle the geometric predicates
```

Comment:
I don't think this will work. If anything, already the input point coordinates would have to be converted into arbitrary-precision numbers. Since the computations with them are slow, first we should try with float64 and if the result is not reliable, resort to big.

The predicates should be really factored out into its own type that satisfies an interface, and carefully and thoroughly tested (random point, co-linear points, almost co-linear points, four points on a circle, four points almost on a circle, etc). Then all geometry-based decisions should be done using these predicates. I think only two are necessary:
```
type OrientationKind int
const (
    CCW OrientationKind = iota
    CW
    Colinear
)

type Predicate interface {
    // Orientation returns how the point (x3,y3) is oriented with respect to
    // the line defined by the points (x1,y1) and (x2,y2).
    Orientation(x1, y1, x2, y2, x3, y3 float64) OrientationKind

    // InCircumcircle returns whether the point (x,y) lies inside
    // the circle defined by the points (x1,y1), (x2,y2), (x3,y3).
    // Maybe a more refined return value could be useful.
    InCircumcircle(x1, y1, x2, y2, x3, y3, x, y float64) bool
}
```
The predicates should be implemented so that eventually the decision boils down to comparing a single number to 0. In case of `Orientation` this would be checking the sign of the determinant
```
|x1 y1 1|
|x2 y2 1|
|x3 y3 1|
```
Here the decisions are complex and spread over many places.

I'm no expert on this and I'm not saying it has to be as I write above but the predicates are the basic building block, they must be robust, and unlike the rest of the algorithm they involve quite a bit of magic.